### ISSUE TITLE
gh-pages seems to be using --global user.email

### ISSUE THREAD

**Issue by kflorence** [tschaub/gh-pages#13]
First off, fantastic module, thanks for it :)

I'm running this command in a repository where my `git config user.email` differs from my global setting, however it seems to be trying to execute commands using my `git config --global user.email` instead, which is causing my pushes to be rejected. Setting my `user.email` to the appropriate email globally fixes the problem.

This may be an id10t error on my part due to some configurations I have locally, but I can't see that I have anything set up wrong, so wanted to see if you could reproduce locally.

**Comment by justinaray** [comment]
@tschaub Perhaps we could get the current value(s) with `git config user.name` and `git config user.email` and use them in the commit?

I am doing this in a script wrapping `gh-pages` and it seems to address this problem.

This could be enabled in the options by something like `user: {useCurrent: true}`

**Comment by tmburnell** [comment]
I ran into this issue also and i believe i have found the issue ... just not sure on the logic yet to fix it. 

One of the first steps is that it steps back a folder and makes a new clone `../.cache`
due to leaving the current project, you lose any local user that you have setup (and it uses global instead).  
Does anyone have any idea of the best place and how to copy the current projects user to the clone (without having to hardcode it in the execution command)?

**Comment by hsribei** [comment]
@justinaray would you mind sharing your wrapper script please? I'm also running into this issue. Thanks!

**Comment by justinaray** [comment]
@hsribei Sorry for the delay.  Been away

Here is an excerpt from my script.  It's dirty, dated, and could definitely be cleaned up with some async/await or other libs.  Nevertheless, it should provide the gist.  Hope it helps.

## tl;dr:

* Read the config via exec
* Specify the user in the `publish` call

## Script Excerpt

```js
const ghpages = require('gh-pages');
const exec = require('child-process-promise').exec;

// ...

let username, email;
exec('git config user.name').then(result => {
    if (!result.stderr) {
        username = result.stdout.trim();
    }

    if (!username) {
        throw new Error('Failed to read user.name');
    }

    return exec('git config user.email');
}).then((result) => {
    if (!result.stderr) {
        email = result.stdout.trim();
    }

    if (!email) {
        throw new Error('Failed to read user.email');
    }

    return;
}).then(() => {
    console.log(`Publishing as ${username}/${email}`);
    ghpages.publish(path.join(__dirname, '../dir/to/push'), {
            // other options
            user: {
                name: username,
                email: email
            }
        }, function (err) {
            if (err) {
                console.error('Failed to publish', err);
                process.exit(1);
            } else {
                console.log('Successfully published');
                process.exit(0);
            }
        }
    );
}).catch(err => {
    console.error(err);
    process.exit(1);
});
```

**Comment by holloway** [comment]
A PR to add local user support as a CLI flag  #229

**Comment by maiermic** [comment]
@tschaub It seems that this issue has been resolved by #264.

**Comment by tschaub** [comment]
Yes, I agree this should have been closed by #264.  Please reopen if there is still an issue.
